lib/router/transition-intent/named-transition-intent.js

```diff
@@ -114,7 +114,11 @@ export default class NamedTransitionIntent extends TransitionIntent {
           newHandlerInfo.context
         );
         var oldContext = oldHandlerInfo && oldHandlerInfo.context;
-        if (result.names.length > 0 && newHandlerInfo.context === oldContext) {
+        if (
+          result.names.length > 0 &&
+          oldHandlerInfo.hasOwnProperty('context') &&
```

Comment:
Does adding this guard allow us to delete this https://github.com/tildeio/router.js/blob/2bfd8e5484f6925d0010777a000782cd84d701ca/lib/router/handler-info.js#L240-L254

If we can do that, we don't need the `hasOwnProperty` check here...

Reply:
We're still going to need `hasOwnProperty` for the scenario where: `oldHandlerInfo.context === undefined && newHandlerInfo.context === undefined` because neither explicitly set it.

It's also theoretically possible that use as a microlib would set context to a "meaningful" `undefined` value on the instance.

Regardless, I believe that `hasOwnProperty` is the correct check here.

Reply:
(See step 7 in https://github.com/tildeio/router.js/pull/242#issuecomment-368662393)